### PR TITLE
fix(preset-built-in): fix mfsu assets mismatch with various publicPat…

### PR DESCRIPTION
**Pull request by xiefengnian**
…h (#7013) (#7044)

* fix(preset-built-in): fix mfsu assets mismatch with various publicPath

* fix(mfsu): make mfsu assets judge more extensible with api

##### Checklist

- [ ] `npm test` passes
- [ ] tests are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

##### Description of change

- any feature?
- close https://github.com/umijs/umi/ISSUE_URL

### CODE REVIEW

File: packages/preset-built-in/src/plugins/features/mfsu/mfsu.ts [hunk 1]
```diff
@@ -256,21 +277,14 @@ export default function (api: IApi) {
   api.addBeforeMiddlewares(() => {
     return (req, res, next) => {
       const path = req.path;
-      // we should remove the first "." to prevent incorrect target such as: ./mf-va_.
-      // because we usually set publicPath: ./ for electron app
-      const publicPath = (api.config.publicPath as string).startsWith('./') ? '/' : api.config.publicPath;
```
Comment (sorrycc) [inline packages/preset-built-in/src/plugins/features/mfsu/mfsu.ts:261]: 这个逻辑和之前的不一样了？

Comment (xiefengnian) [reply packages/preset-built-in/src/plugins/features/mfsu/mfsu.ts:261]: 看用例这个逻辑应该是没问题的

Comment (sorrycc) [reply packages/preset-built-in/src/plugins/features/mfsu/mfsu.ts:261]: @leftstick 确认下这个改动是否会影响到 electron 的使用？

Comment (leftstick) [reply packages/preset-built-in/src/plugins/features/mfsu/mfsu.ts:261]: @sorrycc @xiefengnian 确认了，在electron下也能正常工作。很给力

**Pull request merged**